### DISCUSSION TITLE
Security System

### DISCUSSION THREAD

**Discussion by ghost** [Sunoo/homebridge-camera-ffmpeg#1286] [Ideas]
Hi, 
I was using dedicated plugin for my foscam camera, but it is not maintained for long time, and has some bugs. But it had really nice feature - "Security System" accessory assigned to every camera. So I had possibility to arm/disarm camera, and based on that, move to some preset position. Is it available in homebridge-camera-ffmpeg in any way? Alternatively I can connect with script using ssh/web api to set preset position, but Security System accessory was very useful.

Thanks

**Comment by Sunoo** [comment]
I’m not sure I understand the feature, can you share a link to the plugin you were using?

**Reply by ghost** [reply]
This is the plugin that I am using now:
https://github.com/justinhulin/homebridge-newfoscam#readme
and sample from homekit:
<img width="148" alt="Screenshot 2022-04-12 at 16 02 17" src="https://user-images.githubusercontent.com/44229051/162979997-c59fa8c3-c68e-48b4-b563-0c53bcffbdc6.png">

And configuration responsible for preset position, e.g.:
"armPreset": "Start",
"disarmPreset": "LeftMost",
Start/LeftMost has to be created in camera settings internally

**Reply by Sunoo** [reply]
Interesting, I think that would be best handled by a separate plugin that provides just that device, but it would certainly be possible. I don’t own any Foscams though, so I’m not sure I’d be the one to do it.

**Reply by ghost** [reply]
Thanks for the info, maybe there is some plugin for that. In the meantime I am migrating to your plugin and arm/disarm will do with rest api directly to cameras, using shortcuts. 
I think that this is not only for foscam cameras, but could be useful/usable for many more.

Sample usage: move up camera when you are in home to be not "monitored". Move to specific position to watch your pets when they are in home, move to other to see whole room when you are on vacation etc.

**Reply by Sunoo** [reply]
The issue is that the API used for that feature is likely Foscam specific. If I were going to implement it into this plugin, I’d want to have some way of making it generic enough for many pan/tilt cameras.

**Reply by ghost** [reply]
Probably you are right, but maybe other cameras can use simple commands like foscam, and you could just run it. User would need to put whole command (like in rtsp field):

sample for foscam:
set infraLed to manual mode
http://IP:PORT/cgi-bin/CGIProxy.fcgi?cmd=setInfraLedConfig&mode=1&usr=USERNAME&pwd=PASSWORD

then turn it off
http://IP:PORT/cgi-bin/CGIProxy.fcgi?cmd=closeInfraLed&usr=USERNAME&pwd=PASSWORD

and go to preset position
http://IP:PORT/cgi-bin/CGIProxy.fcgi?cmd=ptzGotoPresetPoint&name=PRESETNAME&usr=USERNAME&pwd=PASSWORD

Maybe in future you could take it into consideration:)

Thanks.
